tests/phpunit/integration/Core/Storage/User_TransientsTest.php

```diff
@@ -22,179 +22,166 @@
  */
 class User_TransientsTest extends TestCase {
 
+	use User_Aware_Interface_ContractTests;
+
 	/**
 	 * @var Context
 	 */
 	protected $context;
 
-	use User_Aware_Interface_ContractTests;
+	/**
+	 * @var string
+	 */
+	protected $old_wp_using_ext_object_cache;
 
-	protected function create_user_aware_instance( $user_id ) {
+	public function setUp() {
+		parent::setUp();
 		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
-		return new User_Transients( $this->context, $user_id );
 	}
 
-	protected function with_predefined_environment( $callback, $ext_object_cache ) {
-		global $_wp_using_ext_object_cache;
+	public function tearDown() {
+		parent::tearDown();
+		wp_using_ext_object_cache( $this->old_wp_using_ext_object_cache );
+	}
 
-		$old_wp_using_ext_object_cache = $_wp_using_ext_object_cache;
-		$_wp_using_ext_object_cache    = 'with_ext_cache' === $ext_object_cache;
+	protected function create_user_aware_instance() {
+		$user_id = $this->factory()->user->create();
 
-		$user_id         = $this->factory()->user->create();
-		$user_transients = $this->create_user_aware_instance( $user_id );
-		call_user_func( $callback, $user_transients, $user_id );
+		return array(
+			new User_Transients( $this->context, $user_id ),
+			$user_id,
+		);
+	}
 
-		$_wp_using_ext_object_cache = $old_wp_using_ext_object_cache;
+	protected function with_predefined_environment( $ext_object_cache ) {
+		$this->old_wp_using_ext_object_cache = wp_using_ext_object_cache( 'with_ext_cache' === $ext_object_cache );
```

Comment:
At first I thought this wouldn't work but the function always returns the previous value when setting it which is nice.

Even though I don't think it is a problem with the current tests, this has the potential to not be restored correctly if this method were to be called more than once.

I appreciate the desire to avoid adding a method for this, but I think the safest and simplest approach here is to capture the previous value once in `wpSetUpBeforeClass` instead, and keep the restore in `tearDown` like you have it.

Reply:
Updated.